### DISCUSSION TITLE
test locally, publish GitHub

### DISCUSSION THREAD

**Discussion by steveschow** [mmistakes/minimal-mistakes#4598] [General]
I am a bit new to **Jekyll** and so please forgive me if my question is dumb.  I am trying to use this theme to start a new website, with no existing content yet.  I want to pretty much use a very similar look and feel as the "docs" section of this theme.

Ultimately I plan to host on GitHub pages and using actions and everything, but what I can't figure out is how to get things working so that I can test it and run it locally and then only push it back to GitHub at certain check points.  As near as I can tell I may have to change some things in the `_config.yml` just prior to pushing it back to GitHub and then locally back to something else in order to work, but I am not completely sure.  Any help on that point would be appreciated. 

The instructions kind of say how to start with nothing, but I want to start with the "docs" model.  I have tried to serve up the docs folder here locally and it doesn't work.. I get the following:

```bash
sjs:[~/mmmore/docs]: bundle exec jekyll serve
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
Configuration file: /Users/sjs/mmmore/docs/_config.yml
            Source: /Users/sjs/mmmore/docs
       Destination: /Users/sjs/mmmore/docs/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
      Remote Theme: Using theme mmistakes/minimal-mistakes
       Jekyll Feed: Generating feed for posts
   GitHub Metadata: No GitHub API authentication could be found. Some fields may be missing or have incorrect data.
                    done in 22.554 seconds.
 Auto-regeneration: enabled for '/Users/sjs/mmmore/docs'
bundler: failed to load command: jekyll (/Users/sjs/.rbenv/versions/3.2.2/bin/jekyll)
<internal:/Users/sjs/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require': cannot load such file -- webrick (LoadError)
```

any ideas at all first how I can get the docs site to run locally...which ideally I'd like to use as a learning model for making my own, since I'm kind of learning how to setup Jekyll sites in general..so need to run that one locally first...and then after that..what will I need to do in order to host it with GitHub Actions, including using the remote gem version of the theme.

later on I will remove all the posts, pages and collections and make my own original content, but first I just want to get it working in the right way so that I will ultimately be able to work on my content locally and push back to GitHub at stages.  

also I just did a clone of the them project to try to run out of there, is it necessary for me to fork it?

**Comment by iBug** [comment]
See <https://github.com/mmistakes/minimal-mistakes/issues/4622#issuecomment-2071385010>
